### PR TITLE
refactor(proxy/http): create `linkerd-http-version` crate

### PR DESCRIPTION
**Pull request by cratelyn**
this outlines the http version type into its own crate.

reëxports are added to make this a backwards compatible change, i.e. no
changes to the public api of `linkerd-proxy-http` are performed here.

### CODE REVIEW

File: linkerd/http/version/src/lib.rs
Comment (cratelyn) [inline linkerd/http/version/src/lib.rs:1]: some brief doc comments are added, while we're here.

File: linkerd/proxy/http/src/lib.rs [hunk 1]
```diff
@@ -50,6 +48,7 @@ pub use http::{
 pub use hyper::body::HttpBody;
 pub use linkerd_http_box::{BoxBody, BoxRequest, BoxResponse, EraseResponse};
 pub use linkerd_http_executor::TracingExecutor;
+pub use linkerd_http_version::{self as version, Version};
```
Comment (cratelyn) [inline linkerd/proxy/http/src/lib.rs:51]: reëxports are added to make this a backwards compatible change, i.e. no
changes to the public api of linkerd-proxy-http are performed here.

File: linkerd/proxy/http/Cargo.toml [hunk 2]
```diff
@@ -42,9 +42,10 @@ linkerd-detect = { path = "../../detect" }
 linkerd-duplex = { path = "../../duplex" }
 linkerd-error = { path = "../../error" }
 linkerd-http-box = { path = "../../http/box" }
+linkerd-http-classify = { path = "../../http/classify" }
 linkerd-http-executor = { path = "../../http/executor" }
 linkerd-http-h2 = { path = "../../http/h2" }
-linkerd-http-classify = { path = "../../http/classify" }
```
Comment (cratelyn) [inline linkerd/proxy/http/Cargo.toml:47]: sorting, while we're here

**Comment by cratelyn** [comment]
:exclamation: enabled auto merge

**Review by olix0r — APPROVED** [review]

**Pull request merged**